code/modules/food_and_drinks/kitchen_machinery/microwave.dm

```diff
@@ -168,6 +168,10 @@
 	user.set_machine(src)
 	interact(user)
 
+/obj/machinery/microwave/AltClick(mob/user)
+	if(user.canUseTopic(src))
```

Comment:
you want `user.canUseTopic(src, be_close=TRUE)`, otherwise you're adding a remote control for microwaves